### PR TITLE
Add export-to-spreadsheet action-button and functionality

### PR DESCRIPTION
**Pull request by gathineou**
Added/edited:

- **query.as_spreadsheet**
- Minor fix in limits/offsets in **query.to_sql**
- Export action-button and JS modal (like csv)
- Google authorization at **views.oauth2_export_to_spreadsheet**
- Callback function at **views.oauth2callback**

Feature is enabled by initializing (patching) the config.google_oauth2_client_secrets_file with oauth-client credentials. 
For setup such credentials check: https://github.com/googleapis/google-api-python-client/issues/678

### CODE REVIEW

File: data_sets/config.py [hunk 1]
```diff
@@ -13,4 +13,14 @@ def data_sets() -> ['data_set.DataSet']:
 
 def charts_color() -> str:
     """The color (rgb hex code) to be used in charts"""
-    return '#008000'
\ No newline at end of file
+    return '#008000'
+
+
+def google_oauth2_client_secrets_file():
```
Comment (jankatins) [inline data_sets/config.py]: We run in docker and for us it's easier to pass in stuff via env vars. See eg the config in the google sheet downloader.

File: data_sets/query.py [hunk 2]
```diff
@@ -186,6 +187,45 @@ def as_csv(self, delimiter, decimal_mark, include_personal_data):
...
+                        row_list.append((list_value_str[:48995] + ' ... ') if len(list_value_str) > 50000 else value)
+                    elif isinstance(value, list):
+                        list_value_str = str(value).replace('\t', ' - ') if len(value) > 0 else ''
+                        row_list.append(
+                            (list_value_str[:48995] + ' ... ') if len(list_value_str) > 50000 else list_value_str)
+                    elif isinstance(value, datetime.datetime):
+                        row_list.append(str(value.strftime("%d-%m-%Y")))
+                    else:
+                        row_list.append(value)
+                data.append(row_list)
```
Comment (jankatins) [inline data_sets/query.py]: No chance of giving the uploader a generator? This might blow up memory on wide, text heavy data sets...

**Review by jankatins — COMMENTED** [review]
I would prefer string based config instead of file names.

Otherwise: looks good!

File: data_sets/query.py (hunk 2 above)
Comment (gathineou) [reply data_sets/query.py]: Thanks for the review Jan! I adjusted the functionality taking in account your comments. Latest commit here: https://github.com/mara/data-sets/commit/2bbb39cb3f30bcf2f88f175b421830ab771efab0

**Pull request merged**